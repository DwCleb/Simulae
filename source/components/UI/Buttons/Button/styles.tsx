import styled from 'styled-components/native';
import { Platform, I18nManager } from 'react-native';
import IonIcon from 'react-native-vector-icons/Ionicons';

import { Metrics, ThemeType } from '~/theme';

const { isRTL } = I18nManager;

interface ButtonProps {
  accessibilityLabel: string;
  style: object;
  opacity: number;
  variant: string;
  disabled: boolean;
  small: boolean;
  distance?: number[];
  background?: string;
}
type Props = ButtonProps & { theme: ThemeType };

function getBackgroundColor({ disabled, theme, variant }: Props): string {
  const primary = theme.button.background.primary.active;
  const primaryDisabled = theme.button.background.primary.disabled;

  const primaryOutlined = theme.button.background.primaryOutlined.active;
  const primaryOutlinedDisabled = theme.button.background.primaryOutlined.disabled;

  const secondary = theme.button.background.secondary.active;
  const secondaryDisabled = theme.button.background.secondary.disabled;

  const secondaryOutlined = theme.button.background.secondaryOutlined.active;
  const secondaryOutlinedDisabled = theme.button.background.secondaryOutlined.disabled;

  if (disabled) {
    switch (variant) {
      case 'primary':
        return primaryDisabled;
      case 'secondaryDisabled':
        return secondaryDisabled;
      case 'secondaryOutlinedDisabled':
        return secondaryOutlinedDisabled;
      case 'primaryOutlinedDisabled':
        return primaryOutlinedDisabled;
      default:
        return '';
    }
  } else {
    switch (variant) {
      case 'primary':
        return primary;
      case 'secondary':
        return secondary;
      case 'secondaryOutlined':
        return secondaryOutlined;
      case 'primaryOutlined':
        return primaryOutlined;
      default:
        return '';
    }
  }
}

function getBorderColor({ disabled, theme, variant }: Props): string {
  const primary = theme.button.border.primary.active;
  const primaryDisabled = theme.button.border.primary.disabled;

  const primaryOutlined = theme.button.border.primaryOutlined.active;
  const primaryOutlinedDisabled = theme.button.border.primaryOutlined.disabled;

  const secondary = theme.button.border.secondary.active;
  const secondaryDisabled = theme.button.border.secondary.disabled;

  const secondaryOutlined = theme.button.border.secondaryOutlined.active;
  const secondaryOutlinedDisabled = theme.button.border.secondaryOutlined.disabled;

  if (disabled) {
    switch (variant) {
      case 'primary':
        return primaryDisabled;
      case 'secondaryDisabled':
        return secondaryDisabled;
      case 'secondaryOutlinedDisabled':
        return secondaryOutlinedDisabled;
      case 'primaryOutlinedDisabled':
        return primaryOutlinedDisabled;
      default:
        return '';
    }
  } else {
    switch (variant) {
      case 'primary':
        return primary;
      case 'secondary':
        return secondary;
      case 'secondaryOutlined':
        return secondaryOutlined;
      case 'primaryOutlined':
        return primaryOutlined;
      default:
        return '';
    }
  }
}

function getIitleColor({ disabled, theme, variant }: Props): string {
  const primary = theme.button.title.primary.active;
  const primaryDisabled = theme.button.title.primary.disabled;

  const primaryOutlined = theme.button.title.primaryOutlined.active;
  const primaryOutlinedDisabled = theme.button.title.primaryOutlined.disabled;

  const secondary = theme.button.title.secondary.active;
  const secondaryDisabled = theme.button.title.secondary.disabled;

  const secondaryOutlined = theme.button.title.secondaryOutlined.active;
  const secondaryOutlinedDisabled = theme.button.title.secondaryOutlined.disabled;

  if (disabled) {
    switch (variant) {
      case 'primary':
        return primaryDisabled;
      case 'secondaryDisabled':
        return secondaryDisabled;
      case 'secondaryOutlinedDisabled':
        return secondaryOutlinedDisabled;
      case 'primaryOutlinedDisabled':
        return primaryOutlinedDisabled;
      default:
        return '';
    }
  } else {
    switch (variant) {
      case 'primary':
        return primary;
      case 'secondary':
        return secondary;
      case 'secondaryOutlined':
        return secondaryOutlined;
      case 'primaryOutlined':
        return primaryOutlined;
      default:
        return '';
    }
  }
}

function getHeight({ small }: Props): number {
  if (small) {
    return 38;
  } else {
    return 50;
  }
}

function getWidth({ small }: Props): number {
  if (small) {
    return Metrics.screen.width * 0.45;
  } else {
    return Metrics.screen.width * 0.9;
  }
}

function getMargin({ distance }: Props): string {
  if (!distance) {
    return `${10}px ${5}px ${10}px ${5}px`;
  } else {
    switch (distance.length || 0) {
      case 1:
        return `${distance[0]}px`;
      case 2:
        return `${distance[0]}px ${distance[1]}px`;
      case 3:
        return `${distance[0]}px ${distance[1]}px ${distance[2]}px`;
      case 4:
        return `${distance[0]}px ${distance[1]}px ${distance[2]}px ${distance[3]}px`;
      default:
        return '0px';
    }
  }
}

export const ActivityIndicator = styled.ActivityIndicator.attrs(({ disabled, theme, variant }: Props) => ({
  color: getIitleColor({ disabled, theme, variant }),
  size: 'large',
}))``;

export const RowView = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

export const IOSCustomButton = styled.TouchableOpacity.attrs((props: Props) => ({
  activeOpacity: props.opacity,
  accessibilityLabel: props.accessibilityLabel,
  style: props.style,
}))`
  flex-direction: row;
  align-items: center;
  height: ${(props: Props): number => getHeight(props)};
  justify-content: center;
  border-width: ${1};
  border-radius: ${8};
  width: ${(props: Props): number => getWidth(props)};
  border-color: ${(props: Props): string => getBorderColor(props)};
  margin: ${(props: Props): string => getMargin(props)};
  background-color: ${(props: Props): string => getBackgroundColor(props)};
  align-self: center;
`;

export const AndroidCustomButton = styled.TouchableNativeFeedback.attrs((props: Props) => ({
  accessibilityLabel: props.accessibilityLabel,
  style: props.style,
}))``;

export const AndroidCustomContentButton = styled.View.attrs((props: Props) => ({
  background: props.background,
  accessibilityLabel: props.accessibilityLabel,
  style: props.style,
}))`
  flex-direction: row;
  align-items: center;
  height: ${(props: Props): number => getHeight(props)};
  justify-content: center;
  border-width: ${1};
  border-radius: ${8};
  width: ${(props: Props): number => getWidth(props)};
  border-color: ${(props: Props): string => getBorderColor(props)};
  margin: ${(props: Props): string => getMargin(props)};
  background-color: ${(props: Props): string => getBackgroundColor(props)};
  align-self: center;
`;

export const TitleButton = styled.Text.attrs((props: Props) => ({
  style: props.style,
  numberOfLines: 1,
}))`
  color: ${(props: Props): string => getIitleColor(props)};
  font-size: 14px;
  line-height: 26px;
  text-transform: uppercase;
  letter-spacing: 1.4px;
`;

interface IconProps {
  name: string;
  logo: boolean;
  ios: boolean;
  md: boolean;
  iconSize: boolean;
  noRTL: boolean;
  iconColor: string;
  direction: string;
}

const platform = Platform.OS === 'ios' ? 'ios' : 'md';

export const Icon = styled(IonIcon).attrs(({ name, logo, ios, md, iconSize, noRTL }: IconProps) => {
  const platformOS = ios ? 'ios' : md ? 'md' : logo ? 'logo' : platform;

  const iconName = `${platformOS}-${name}`;

  return {
    name: iconName,
    logo,
    styleIcon: { transform: [{ scaleX: !noRTL && isRTL ? -1 : 1 }] },
    size: iconSize || 22,
  };
})`
  color: ${(props: IconProps): string => props.iconColor || 'white'};
  margin-left: ${(props: IconProps): number =>
    props.direction === 'start' ? 8 : Platform.select<number>({ android: 10, ios: 8 }) || 8};
  margin-right: ${(props: IconProps): number => {
    const marginRight =
      props.direction === 'start'
        ? Platform.select<number>({ android: 10, ios: 8 })
        : Platform.select<number>({ android: 1, ios: 8 });

    return marginRight || 8;
  }};
`;

export const rippleOverlay = 'rgba(0, 0, 0, 0, 0.6)';
