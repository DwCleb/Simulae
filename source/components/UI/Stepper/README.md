```jsx
// TODO: create a Wrapper to react-styleguidist to provide theme to all components shown
import ThemeManager from '../../../theme/ThemeContext';
import Stepper from '.';

import { View } from 'react-native';

<ThemeManager>
  <View style={{ height: 50 }}>
    <Stepper length={7} acticeDotIndex={3} />
  </View>
</ThemeManager>;
```
