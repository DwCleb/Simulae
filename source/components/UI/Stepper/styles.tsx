import styled from 'styled-components/native';
import { Pagination } from 'react-native-snap-carousel';

import { Colors, ThemeContextType } from '~/theme';

import { StepperProps } from './index';

const { blue, red, orange } = Colors.theme.primary;

const { green, yellow } = Colors.theme.alerts;

export const SteppersColors = [blue, red, orange, green, yellow];

export const StepperLength = SteppersColors.length;

// FIXME: this is a workaround, we should use StyleComponents types
type ThemedStepperProps = StepperProps & ThemeContextType;

export const Steppers = styled(Pagination).attrs(({ activeDotIndex, theme }: ThemedStepperProps) => ({
  dotColor: SteppersColors[activeDotIndex >= 7 ? activeDotIndex - StepperLength : activeDotIndex],
  dotStyle: {
    width: 35,
    height: 10,
    borderRadius: 50,
    shadowColor: '#000',
    shadowOffset: [0, -8],
    shadowOpacity: 0.2,
    shadowRadius: 8,
    elevation: 6,
  },
  inactiveDotStyle: {
    width: 10,
    height: 10,
    borderRadius: 50,
    shadowColor: '#000',
    shadowOffset: [0, -8],
    shadowOpacity: 0.3,
    shadowRadius: 8,
    elevation: 6,
  },
  containerStyle: { paddingTop: 20 },
  inactiveDotColor: theme.monochromatic.white,
}))``;

const plansStepperColors = [orange, blue, yellow];

export const PlansSteppers = styled(Pagination).attrs(({ activeDotIndex, theme }: ThemedStepperProps) => ({
  dotColor: plansStepperColors[activeDotIndex >= 3 ? activeDotIndex - StepperLength : activeDotIndex],
  dotStyle: {
    width: 20,
    height: 5,
    borderRadius: 50,
  },
  inactiveDotStyle: {
    width: 5,
    height: 5,
    borderRadius: 50,
  },
  containerStyle: { position: 'absolute', bottom: 0, alignSelf: 'center' },
  inactiveDotColor: theme.monochromatic.white,
}))``;
