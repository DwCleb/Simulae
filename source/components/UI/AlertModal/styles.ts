import styled from 'styled-components/native';
import Modal from 'react-native-modal';
import { ThemeContextType } from '~/theme';
import { Typography } from '~/components';
import { TypographyType } from '~/components/UI/Typography';

export const Container = styled(Modal)`
  flex: 1;
  justify-content: center;
  margin: 20px;
  background-color: transparent;
  padding: 10px;
`;

export const Content = styled.View`
  width: 100%;
  background-color: white;
  border-radius: 15px;
`;

export const Title = styled(Typography).attrs(() => ({
  type: TypographyType.H3,
}))`
  padding: 20px;
  color: ${({ theme }: ThemeContextType): string => theme.monochromatic.black};
`;

export const Message = styled(Typography).attrs(() => ({
  type: TypographyType.P2,
}))`
  padding-bottom: 20px;
  color: ${({ theme }: ThemeContextType): string => theme.monochromatic.darkGrey};
`;

export const Button = styled.TouchableOpacity.attrs(() => ({
  activeOpacity: 0.6,
}))`
  padding: 15px;
  border-bottom-left-radius: 15px;
  border-bottom-right-radius: 15px;
  background-color: ${({ theme }: ThemeContextType): string => theme.alerts.green};
`;

export const ButtonText = styled(Typography).attrs(() => ({
  type: TypographyType.P1,
  bold: true,
}))`
  color: ${({ theme }: ThemeContextType): string => theme.monochromatic.black};
`;
