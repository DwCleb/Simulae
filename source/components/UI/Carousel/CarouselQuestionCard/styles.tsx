import styled from 'styled-components/native';
import { SteppersColors, StepperLength } from '~/components/UI/Stepper/styles';

export const Container = styled.View`
  /* flex: 1; */
  height: 70%;
  background-color: white;
  shadow-color: #000;
  shadow-offset: 0px -8px;
  shadow-opacity: 0.25;
  shadow-radius: 8px;
  elevation: 6;
  justify-content: center;
  align-items: center;
  border-radius: 12px;
  border-top-color: ${({ cardIndex }): string =>
    SteppersColors[cardIndex >= 7 ? cardIndex - StepperLength : cardIndex] || 'red'};
  border-top-width: 10px;
`;

export const TitleContainer = styled.View`
  justify-content: center;
  align-items: center;
`;

export const IllustrationContainer = styled.View`
  justify-content: center;
  align-items: center;
`;

export const DescriptionContainer = styled.View`
  justify-content: center;
  align-items: center;
`;

export const CarouselImage = styled.Image.attrs(() => ({
  resizeMode: 'cover',
}))`
  border-radius: 80px;
  /* margin: ${20}px 0px; */
  align-self: center;
  width: 100px;
  height: 100px;
  /* width: ${({ width }): number | string => (typeof width === 'number' ? `${width}px` : width)};
  height: ${({
    height,
  }): number | string => (typeof height === 'number' ? `${height}px` : height)}; */
`;
