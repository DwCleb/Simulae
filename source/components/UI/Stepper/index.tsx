import React from 'react';

import { Steppers, PlansSteppers } from './styles';

export interface StepperProps {
  length?: number;
  activeDotIndex: number;
  templateStepper?: string;
}

export default function Stepper({ templateStepper, length, activeDotIndex, ...props }: StepperProps): JSX.Element {
  switch (templateStepper) {
    case 'plansStepper':
      return (
        <PlansSteppers
          dotsLength={length}
          activeDotIndex={activeDotIndex}
          inactiveDotOpacity={1}
          inactiveDotScale={0.9}
          {...props}
        />
      );

    default:
      return (
        <Steppers
          dotsLength={length}
          activeDotIndex={activeDotIndex}
          inactiveDotOpacity={1}
          inactiveDotScale={0.9}
          {...props}
        />
      );
  }
}
