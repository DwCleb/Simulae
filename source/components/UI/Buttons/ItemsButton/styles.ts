import { StyleSheet } from 'react-native';
import styled from 'styled-components/native';

import { Typography } from '~/components';
import { TypographyType } from '~/components/UI/Typography';

export const styles = StyleSheet.create({
  button: {
    width: 100,
    height: 100,
    borderRadius: 10,
    backgroundColor: '#38CFEE',
    margin: 10,
  },

  image: {
    position: 'relative',
    height: 100,
    width: 100,
    borderBottomLeftRadius: 10,
    borderBottomRightRadius: 10,
    overflow: 'hidden',
  },
});

export const ItemName = styled(Typography).attrs(() => ({
  type: TypographyType.P2,
}))`
  color: ${({ theme }): string => theme.monochromatic.white};
  text-align: center;
`;

export const ItemNumber = styled(Typography).attrs(() => ({
  type: TypographyType.P2,
}))`
  color: ${({ theme }): string => theme.monochromatic.black};
  text-align: right;
  margin-right: 6px;
`;
