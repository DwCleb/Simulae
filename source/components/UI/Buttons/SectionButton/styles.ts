import styled from 'styled-components/native';
import { Colors } from '~/theme';
import { Typography } from '~/components';
import { TypographyType } from '~/components/UI/Typography';
import IonIcon from 'react-native-vector-icons/Ionicons';

export const Container = styled.TouchableOpacity.attrs(() => ({
  activeOpacity: 0.6,
}))`
  width: 100%;
  height: 60px;
  background-color: ${Colors.theme.monochromatic.white};
  padding: 20px;
  justify-content: center;
  flex-direction: row;
`;

export const Title = styled(Typography).attrs(() => ({
  type: TypographyType.P1,
  align: 'left',
}))`
  color: ${({ theme }): string => theme.monochromatic.darkGrey};
  align-self: center;
  position: absolute;
  left: 20px;
`;

export const Value = styled(Typography).attrs(() => ({
  type: TypographyType.L1,
  align: 'right',
}))`
  align-self: center;
  position: absolute;
  right: 60px;
  color: ${({ theme }): string => theme.monochromatic.darkGrey};
`;

export const RightArrow = styled(IonIcon).attrs((props) => ({
  size: props.size || 30,
  color: Colors.theme.monochromatic.darkGrey,
  name: 'chevron-forward',
}))`
  align-self: center;
  position: absolute;
  right: 20px;
  color: ${({ theme }): string => theme.monochromatic.darkGrey};
`;
