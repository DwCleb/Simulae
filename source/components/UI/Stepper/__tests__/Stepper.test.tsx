import React from 'react';
import { create, ReactTestRenderer } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import Stepper, { StepperProps } from '~/components/UI/Stepper';

import darkTheme from '~/theme/ThemeContext/dark';
import { Steppers, PlansSteppers } from '../styles';

let wrapper: ReactTestRenderer;
let root: ReactTestRenderer['root'];

const props: StepperProps = { length: 8, activeDotIndex: 8 };

const theme = darkTheme.theme;

const componentToTest = (componentProps: StepperProps): JSX.Element => (
  <ThemeProvider theme={theme}>
    <Stepper {...componentProps} />
  </ThemeProvider>
);

describe('<Stepper />', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  beforeEach(() => {
    wrapper = create(componentToTest(props));
    root = wrapper.root;
  });

  it('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render the correct structure with the default props', () => {
    expect(root.findAllByType(Steppers)).toHaveLength(1);

    expect(root.findAllByType(Steppers)[0].props.dotsLength).toEqual(props.length);
    expect(root.findAllByType(Steppers)[0].props.activeDotIndex).toEqual(props.activeDotIndex);
  });

  it('Should render correctly with plansStepper template', () => {
    const newProps = { ...props, templateStepper: 'plansStepper' };

    wrapper = create(componentToTest(newProps));
    root = wrapper.root;

    expect(root.findAllByType(PlansSteppers)).toHaveLength(1);

    expect(root.findAllByType(PlansSteppers)[0].props.dotsLength).toEqual(props.length);
    expect(root.findAllByType(PlansSteppers)[0].props.activeDotIndex).toEqual(props.activeDotIndex);
  });

  it('Should render correctly with wizard template', () => {
    const newProps = { ...props, templateStepper: 'wizard' };

    wrapper = create(componentToTest(newProps));
    root = wrapper.root;

    expect(root.findAllByType(Stepper)[0].props.children).toBe(undefined);
  });

  it('Should render correctly with subwizardPrimary template', () => {
    const newProps = { ...props, templateStepper: 'subwizardPrimary' };

    wrapper = create(componentToTest(newProps));
    root = wrapper.root;

    expect(root.findAllByType(Stepper)[0].props.children).toBe(undefined);
  });

  it('Should render correctly with subwizardSecondary template', () => {
    const newProps = { ...props, templateStepper: 'subwizardSecondary' };

    wrapper = create(componentToTest(newProps));
    root = wrapper.root;

    expect(root.findAllByType(Stepper)[0].props.children).toBe(undefined);
  });
});
