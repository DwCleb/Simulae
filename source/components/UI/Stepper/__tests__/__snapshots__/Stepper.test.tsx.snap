// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stepper /> Should render correctly 1`] = `
<View
  accessible={false}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 30,
      },
      Object {
        "flexDirection": "row",
      },
      Object {
        "paddingTop": 20,
      },
    ]
  }
>
  <View
    accessible={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 100,
          "height": 20,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 0.9,
            },
          ],
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 100,
          "height": 20,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 0.9,
            },
          ],
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 100,
          "height": 20,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 0.9,
            },
          ],
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 100,
          "height": 20,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 0.9,
            },
          ],
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 100,
          "height": 20,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 0.9,
            },
          ],
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 100,
          "height": 20,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 0.9,
            },
          ],
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 100,
          "height": 20,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 0.9,
            },
          ],
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 100,
          "height": 20,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 0.9,
            },
          ],
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
